homeassistant/components/izone/climate.py

```diff
@@ -388,49 +389,45 @@ def max_temp(self) -> float:
         """Return the maximum temperature."""
         return self._controller.temp_max
 
-    async def wrap_and_catch(self, coro):
-        """Catch any connection errors and set unavailable."""
-        try:
-            await coro
-        except ConnectionError as ex:
-            self.set_available(False, ex)
-        else:
-            self.set_available(True)
-
     async def async_set_temperature(self, **kwargs: Any) -> None:
         """Set new target temperature."""
         if not self.supported_features & ClimateEntityFeature.TARGET_TEMPERATURE:
             self.async_schedule_update_ha_state(True)
             return
         if (temp := kwargs.get(ATTR_TEMPERATURE)) is not None:
-            await self.wrap_and_catch(self._controller.set_temp_setpoint(temp))
+            with suppress(ConnectionError):
+                await self._controller.set_temp_setpoint(temp)
 
     async def async_set_fan_mode(self, fan_mode: str) -> None:
         """Set new target fan mode."""
         fan = self._fan_to_pizone[fan_mode]
-        await self.wrap_and_catch(self._controller.set_fan(fan))
+        with suppress(ConnectionError):
+            await self._controller.set_fan(fan)
```

Comment:
Command/service errors, should not be suppressed.
Instead, it should be caught and handled.

Reply:
In that case it would be best to close the PR as this was already being done.

The disconnection is handled elsewhere by a listener that listens for the disconnection. The error here can safely be ignored as its attempting to send state when the service is already unavailable (and is marked as unavailable) which shouldn't happen in general use (unless there's a race) and if it does it's not important to raise exceptions about it as it would just clutter up the log for no purpose.

Could just do a try/catch and ignore the error alternately, but that's syntactically the same.

Reply:
That wasn't done either. The error needs to be wrapped in an `HomeAssistantError` bubbled up. This way, e.g, automation using it will fail (and show up as such in the traces as well).

Otherwise, the automation just silently pass and the user has no idea why it didn't work.

Reply:
OK will fix

Reply:
Ok that should be better. Am raising exceptions like suggested. 

What about the property getters when offline? Have returned a default value, mostly `None`. but depending on what's most appropriate.

Reply:
@frenck Will do some more testing tomorrow with broken connections before ready to mark as ready, but yeh the concept would be good to review

Reply:
@frenck So I did a bit more testing. The service calls don't even seem to get made if the controller is disconnected and marked unavailable. The exceptions therefore only get raised when the service goes offline, and  there's a race between the service call being made and, device being marked unavailable with the service call winning. It seems a bit inconsistent to *sometimes* raise exceptions? Is that really right? sorry a bit confused about this.

I had a look at hue for an example and it's doing the same as what I've re-coded this as now and as recommended.

Reply:
@frenck good to go? Need this to complete #89730